src/nodedef.cpp

```diff
@@ -41,6 +41,45 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include <algorithm>
 #include <cmath>
 
+
+// Should be ordered exactly like enum NodeDrawType in nodedef.h
+struct EnumString INodeDefManager::DrawTypes[] =
+	{
+		{NDT_NORMAL, "normal"},
+		{NDT_AIRLIKE, "airlike"},
+		{NDT_LIQUID, "liquid"},
+		{NDT_FLOWINGLIQUID, "flowingliquid"},
+		{NDT_GLASSLIKE, "glasslike"},
+		{NDT_GLASSLIKE_FRAMED, "glasslike_framed"},
+		{NDT_GLASSLIKE_FRAMED_OPTIONAL, "glasslike_framed_optional"},
+		{NDT_ALLFACES, "allfaces"},
+		{NDT_ALLFACES_OPTIONAL, "allfaces_optional"},
+		{NDT_TORCHLIKE, "torchlike"},
+		{NDT_SIGNLIKE, "signlike"},
+		{NDT_PLANTLIKE, "plantlike"},
+		{NDT_FIRELIKE, "firelike"},
+		{NDT_FENCELIKE, "fencelike"},
+		{NDT_RAILLIKE, "raillike"},
+		{NDT_NODEBOX, "nodebox"},
+		{NDT_MESH, "mesh"},
+		{NDT_PLANTLIKE_ROOTED, "plantlike_rooted"},
+		{0, NULL},
+	};
+
+bool INodeDefManager::string_to_enum (const EnumString *spec, int &result, const std::string &drawtype) const
```

Comment:
Line too long, superfluous space near `(`
This function should be named `stringToEnum` or similar
Low priority: Could you add this function as `inline` into `c_types.h`? It's a helper function for this struct array.